Add alwaysAddToCart behaviour to ProductSummaryBuyButton (#249)
vtex-apps/product-summary: docs/ProductSummaryBuyButton.md

📢 Use this project, [contribute](https://github.com/vtex-apps/product-summary) to it or open issues to help evolve it using [Store Discussion](https://github.com/vtex-apps/store-discussion).

# Product Summary Buy Button

> `ProductSummaryBuyButton` is a VTEX Component that renders the buy button.

Author: ```suggestion
Product Summary Buy Button is a block exported by the Product Summary app responsible for rendering a buy button in the Product Summary Shelf block.
```

> This Component can be imported and used by any VTEX App.

Reviewer: Instead of this sentence, I strongly recommend to add a media here (Buy Button being displayed)

  }
```

2. Add `product-summary-buy-button` anywhere inside the `product-summary.shelf`:

```diff
 {
   "product-summary.shelf": {
       "children": [
           "product-summary-image",
           "product-summary-name",
           "product-rating-inline",
           "product-summary-space",
           "product-summary-price",
+          "product-summary-buy-button"
       ]
   },
 }
```

### Props

| Prop name           | Type      | Description                                                                                 | Default value         |
| ------------------- | --------- | ------------------------------------------------------------------------------------------- | --------------------- |
| `isOneClickBuy`     | `Boolean` | Should redirect to checkout after clicking on buy                                           | `false`               |
| `buyButtonText`     | `String`  | Custom buy button text                                                                      |                       |
| `displayBuyButton`  | `Enum`    | Set display mode of buy button (displayButtonAlways, displayButtonHover, displayButtonNone) | `displayButtonAlways` |

> | `customToastURL`  | `String`    | Set the link associated with the Toast created when adding an item to your cart.  | `/checkout/#/cart` |

Reviewer: | `customToastURL`  | `String`    | Defines a redirect link to the Toast displayed when an item is added to your cart.  | `/checkout/#/cart` |

> | `buyButtonBehavior` | `Enum` | What will happen when the user clicks on the button. Check the options bellow. | `default`

Reviewer: | `buyButtonBehavior` | `Enum` | Sets the Buy Button behavior when it is clicked on. You can choose between the following scenarios: <ul><li>`alwaysGoToProduct` - Redirect users to the product page;</li><li>`default` - Redirect users to the Product Page when there are several SKUs available. In scenarios in which there is only one SKU available, it will be add to the cart ;</li><li>`alwaysAddToTheCart` - Always add the selected SKU to the cart. When choosing this option, be careful: use it only if there are SKU Selectors for each product variation, so users can properly select their desired SKU. </li></ul> | `default` |

- Possible values for `buyButtonBehavior`:

| Value | Description |
| --- | --- |

> | `default` | It will add to cart if the product has one SKU. It will link to the product page if it has more than one SKU. |

Reviewer: no need. The content is already in the table above.

> | `alwaysGoToProduct` | It will always link to the product page. |

Reviewer: no need. The content is already in the table above.

> | `alwaysAddToCart` | It will always add the selected item to cart. **Be careful**: use this option only if the user is able to select all variations of the product. Example: if the product has Color and Size, you should show the SKU Selector of both variations so the user can select the color and size they want, otherwise the button will add a product that the user was not able to choose, for example, the first Size option. |

Reviewer: no need. The content is already in the table above.

+          "product-summary-buy-button"
       ]
   },
 }
```

### Props

| Prop name           | Type      | Description                                                                                 | Default value         |
| ------------------- | --------- | ------------------------------------------------------------------------------------------- | --------------------- |
| `isOneClickBuy`     | `Boolean` | Should redirect to checkout after clicking on buy                                           | `false`               |
| `buyButtonText`     | `String`  | Custom buy button text                                                                      |                       |
| `displayBuyButton`  | `Enum`    | Set display mode of buy button (displayButtonAlways, displayButtonHover, displayButtonNone) | `displayButtonAlways` |
| `customToastURL`  | `String`    | Set the link associated with the Toast created when adding an item to your cart.  | `/checkout/#/cart` |
| `buyButtonBehavior` | `Enum` | What will happen when the user clicks on the button. Check the options bellow. | `default`

- Possible values for `buyButtonBehavior`:

| Value | Description |
| --- | --- |

> | `default` | It will add to cart if the product has one SKU. It will link to the product page if it has more than one SKU. |

Reviewer 2: ```suggestion
| `default` | It will add to cart if the product has only one SKU; otherwise it will redirect to the product page. |
```

Author: I prefer how it is right now

> | `alwaysGoToProduct` | It will always link to the product page. |

Reviewer 2: ```suggestion
| `alwaysGoToProduct` | It will always redirect to the product page. |
```

> | `alwaysAddToCart` | It will always add the selected item to cart. Be careful: use this option only if the user is able to select all variations of the product. Example: if the product has Color and Size, you should show the SKU Selector of both variations so the user can select the color and size they want, otherwise the button will add a product that the user was not able to choose, for example, the first Size option. |

Reviewer 2: ```suggestion
| `alwaysAddToCart` | It will always add the selected item to cart. **Be careful**: use this option only if the user is able to select all variations of the product. Example: if the product has Color and Size, you should show the SKU Selector of both variations so the user can select the color and size they want, otherwise the button will add a product that the user was not able to choose, for example, the first Size option. |
```
